lib/internal/modules/cjs/loader.js

```diff
@@ -611,9 +619,14 @@ Module._findPath = function(request, paths, isMain) {
 
   // For each path
   for (let i = 0; i < paths.length; i++) {
-    // Don't search further if path doesn't exist and request is inside the path
+    // Don't search further if path doesn't exist
+    // or doesn't have permission to it
     const curPath = paths[i];
-    if (insidePath && curPath && _stat(curPath) < 1) continue;
+    if (insidePath && curPath &&
+      ((permission.isEnabled() && permission.has('fs.read', curPath) === false) || _stat(curPath) < 1)
```

Comment:
no, that's correct. `_stat` performs a file read, so to avoid throwing ERR_ACCESS_DENIED error we need to check if it doesn't have permission.

> (skipping the stat check when permission is allowed seems wrong)

The skip happens when permission is NOT allowed.

Reply:
Yeah sorry, I read the condition again today and it's correct!